Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,236 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var maxBalance = 0m;
+            var name = "";
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var balance = GetLargestBalance(peopleBalances);
+
+                if (balance >= maxBalance)
+                {
+                    if (balance > maxBalance)
+                    {
+                        name = "";
+                    }
+                    if (name != "")
+                    {
+                        name += ", ";
+                    }
+                    name += peopleBalances[0];
+
+                    maxBalance = balance;
+                }
+            }
+
+            SetMoneyFormat();
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            return $"{ReplaceLastComma(name)} had the most money ever. {maxBalanceNoComma}.";
+        }
+
+        private static decimal GetLargestBalance(string[] peopleBalances)
+        {
+            var maxBalance = 0m;
+
+            for (var i = 1; i < peopleBalances.Length; i++)
+            {
+                var isBalanceValid = decimal.TryParse(peopleBalances[i], out var balance);
+
+                if (isBalanceValid && balance > maxBalance)
+                {
+                    maxBalance = balance;
+                }
+            }
+
+            return maxBalance;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var biggestLossEver = decimal.MaxValue;
+            var name = "";
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+
+                if (peopleBalances.Length <= 2)
+                {
+                    continue;
+                }
+
+                var biggestLossForPerson = GetBiggestLossForPerson(peopleBalances);
+
+                if (biggestLossForPerson < biggestLossEver)
+                {
+                    biggestLossEver = biggestLossForPerson;
+                    name = peopleBalances[0];
+                }
+            }
+
+            if (name == "")
+            {
+                return "N/A.";
+            }
+
+            SetMoneyFormat();
+            var biggestLossEverNoComma = RemoveComma($"{biggestLossEver:C0}").Replace("(", "-").Replace(")", ""); ;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4955013)